### PR TITLE
Update configurations / dependencies to the latest versions.

### PR DESCRIPTION
**Pull request by mrvicadai**
## Changelog

- Fix `bsconfig.json` warning by specifying `refmt: 3`
- Use `yarn` as default package manager.
- Update devDependencies to latest versions.
- Default to use `Main.re` instead of `Main.ml`.
- Various formatting.

### CODE REVIEW

**Comment by tcoopman** [comment]
I'm fine with updating the config, but I'm not sure about having a default for `reason`. Can you provide both `ml` and `re` files like in https://github.com/OvermindDL1/bucklescript-tea?

**Comment by mrvicadai** [comment]
@tcoopman thanks for reviewing! Just addressed the comment by reverting to `Main.ml` in 
https://github.com/tcoopman/bucklescript-tea-starter-kit/pull/3/commits/a2ce86513366d132e574bf63479dd09f9b3b0698

File: .gitignore [hunk 1]
```diff
@@ -25,7 +25,6 @@ lib/
 node_modules
 release/main.js
 .merlin
-yarn.lock
```
Comment (tcoopman) [inline .gitignore:28]: I find this a hard thing and will always be subject to preference, so I'll propose that we keep `npm` for now.

**Review by tcoopman — CHANGES_REQUESTED** [review]
Let's revert the yarn stuff for now (so also the lock files). I think this is personal preference, so not really worth the change.

File: .gitignore (hunk 1 above)
Comment (mrvicadai) [reply .gitignore:28]: Alright, this is addressed in 
https://github.com/tcoopman/bucklescript-tea-starter-kit/pull/3/commits/5122b40bc08a0216d183136f66dbc7a0cf5a52d5

**Review by tcoopman — APPROVED** [review]

**Pull request merged**